### PR TITLE
Add `migrate --describe-json` to output machine-readable migration information

### PR DESCRIPTION
**Pull request by seesemichaelj**
This PR adds a `--describe-json` flag for `truffle migrate` which outputs a standardized format (a JSON object with `status` and `data` fields) for the migration status. This is primarily to make the migration output to be machine-readable/consumable.

Here is an example output for the `metacoin` project:

```
$ truffle-dev migrate --describe-json --reset

Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.

MIGRATION_STATUS:{"status":"preAllMigrations","data":{"dryRun":false,"migrations":[{"file":"/home/mike/work/truffle/metacoin/migrations/1_initial_migration.js","number":1,"isFirst":true,"isLast":false,"dryRun":false,"interactive":false},{"file":"/home/mike/work/truffle/metacoin/migrations/2_deploy_contracts.js","number":2,"isFirst":false,"isLast":true,"dryRun":false,"interactive":false}]}}

Starting migrations...
======================
> Network name:    'development'
> Network id:      1567042575243
> Block gas limit: 0x6691b7

MIGRATION_STATUS:{"status":"preMigrate","data":{"file":"1_initial_migration.js","number":1,"isFirst":true,"network":"development","networkId":1567042575243,"blockLimit":"0x6691b7"}}

1_initial_migration.js
======================
MIGRATION_STATUS:{"status":"replacing","data":{"contractName":"Migrations","priorAddress":"0xD2F26add3a0840911C484EC21e216cEade45b830"}}

   Replacing 'Migrations'
   ----------------------
   > transaction hash:    0xcf60633a81105b18d041c673403450ad1543b1d2675dffb5106088d89e90953e
   > Blocks: 0            Seconds: 0
   > contract address:    0x061b75d1bFb189A1A370AfcDa115eB15A6594AE4
   > block number:        41
   > block timestamp:     1567054697
   > account:             0xfD68706A046aC2CfD876854019824E0291ed2a81
   > balance:             99.88189794
   > gas used:            261393
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00522786 ETH
MIGRATION_STATUS:{"status":"deployed","data":{"contract":{"contractName":"Migrations","address":"0xD2F26add3a0840911C484EC21e216cEade45b830"},"deployed":true,"receipt":{"transactionHash":"0xcf60633a81105b18d041c673403450ad1543b1d2675dffb5106088d89e90953e","gasUsed":261393},"timestamp":1567054697,"gasPrice":"20","gas":"261393","from":"0xfD68706A046aC2CfD876854019824E0291ed2a81","value":"0","cost":"0.00522786","balance":"99.88189794"}}

   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.00522786 ETH

MIGRATION_STATUS:{"status":"postMigrate","data":{"number":1,"cost":"0.00522786"}}

MIGRATION_STATUS:{"status":"preMigrate","data":{"file":"2_deploy_contracts.js","number":2,"isFirst":false,"network":"development","networkId":1567042575243,"blockLimit":"0x6691b7"}}

2_deploy_contracts.js
=====================
MIGRATION_STATUS:{"status":"replacing","data":{"contractName":"ConvertLib","priorAddress":"0x9a091dD594d124496C70299a15336e1368bb9b0F"}}

   Replacing 'ConvertLib'
   ----------------------
   > transaction hash:    0xc0d429c696fa89d09074c72e8c8c57a6f661e6adc0cf3561adee18c560ae5a7e
   > Blocks: 0            Seconds: 0
   > contract address:    0xAE1e789291Ab535a32f5C05C79A86De842a6f9A5
   > block number:        43
   > block timestamp:     1567054697
   > account:             0xfD68706A046aC2CfD876854019824E0291ed2a81
   > balance:             99.87882166
   > gas used:            111791
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00223582 ETH
MIGRATION_STATUS:{"status":"deployed","data":{"contract":{"contractName":"ConvertLib","address":"0x9a091dD594d124496C70299a15336e1368bb9b0F"},"deployed":true,"receipt":{"transactionHash":"0xc0d429c696fa89d09074c72e8c8c57a6f661e6adc0cf3561adee18c560ae5a7e","gasUsed":111791},"timestamp":1567054697,"gasPrice":"20","gas":"111791","from":"0xfD68706A046aC2CfD876854019824E0291ed2a81","value":"0","cost":"0.00223582","balance":"99.87882166"}}

   Linking
   -------
   * Contract: MetaCoin <--> Library: ConvertLib (at address: 0xAE1e789291Ab535a32f5C05C79A86De842a6f9A5)
MIGRATION_STATUS:{"status":"replacing","data":{"contractName":"MetaCoin","priorAddress":"0xA65c5d3bC992edAEB130759b10d85b9d52858724"}}

   Replacing 'MetaCoin'
   --------------------
   > transaction hash:    0x3051a93247226407a9b2344c28cca9a8bf9b223f184f644ec6095c7e048b7be4
   > Blocks: 0            Seconds: 0
   > contract address:    0x583Fbb1bC9cdcee64147E9167A3F96143683c623
   > block number:        44
   > block timestamp:     1567054697
   > account:             0xfD68706A046aC2CfD876854019824E0291ed2a81
   > balance:             99.87382938
   > gas used:            249614
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00499228 ETH
MIGRATION_STATUS:{"status":"deployed","data":{"contract":{"contractName":"MetaCoin","address":"0xA65c5d3bC992edAEB130759b10d85b9d52858724"},"deployed":true,"receipt":{"transactionHash":"0x3051a93247226407a9b2344c28cca9a8bf9b223f184f644ec6095c7e048b7be4","gasUsed":249614},"timestamp":1567054697,"gasPrice":"20","gas":"249614","from":"0xfD68706A046aC2CfD876854019824E0291ed2a81","value":"0","cost":"0.00499228","balance":"99.87382938"}}

   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:           0.0072281 ETH

MIGRATION_STATUS:{"status":"postMigrate","data":{"number":2,"cost":"0.0072281"}}


Summary
=======
> Total deployments:   3
> Final cost:          0.01245596 ETH

MIGRATION_STATUS:{"status":"lastMigrate","data":{"totalDeployments":"3","finalCost":"0.01245596"}}

MIGRATION_STATUS:{"status":"postAllMigrations","data":{"dryRun":false,"error":null}}
```

### CODE REVIEW

**Comment by eggplantzzz** [comment]
The comments of mine are not blocking this from moving forward, they are just my own curiosity.

**Comment by seesemichaelj** [comment]
@CruzMolina The tests were added seemed to pass. The failing check is due to other issues not related to this PR I believe. Is this sufficient?

File: packages/reporters/reporters/migrations-V5/messages.js [hunk 1]
```diff
@@ -230,6 +283,23 @@ class MigrationsMessages {
             `Pausing for ${reporter.confirmations} confirmations...`
           );
 
+        if (this.describe) {
```
Comment (gnidan) [inline packages/reporters/reporters/migrations-V5/messages.js]: Could we call this `--describe-json`?

Comment (seesemichaelj) [reply packages/reporters/reporters/migrations-V5/messages.js]: sure

**Pull request merged**